mobile/src/reducers/calendar/calendar.epics.ts

```diff
@@ -50,4 +50,34 @@ export const intervalsBySingleDaySelectionEpic$ = (action$: ActionsObservable<Se
 
 export const calendarSelectionModeEpic$ = (action$: ActionsObservable<CalendarSelectionMode>) =>
     action$.ofType('CALENDAR-SELECTION-MODE')
-        .map(x => disableCalendarSelection(false));
\ No newline at end of file
+        .map(x => disableCalendarSelection(false));
+
+export const calendarEventApprovedEpic$ = (action$: ActionsObservable<CalendarEventApprove>, state: AppState, deps: DependenciesContainer) =>
+    action$.ofType('CALENDAR-EVENT-APPROVE')
+        .flatMap(x => {
+            const requestBody = { ...x.calendarEvent };
+
+            requestBody.status = CalendarEventStatus.Approved;
+
+            return deps.apiClient.put(
+                `/employees/${x.employeeId}/events/${x.calendarEvent.calendarEventId}`,
+                requestBody,
+                { 'Content-Type': 'application/json' }
+            ).map(obj => loadCalendarEvents(x.employeeId));
+        })
+        .catch((e: Error) => Observable.of(loadFailedError(e.message)));
+
+export const calendarEventRejectedEpic$ = (action$: ActionsObservable<CalendarEventReject>, state: AppState, deps: DependenciesContainer) =>
+    action$.ofType('CALENDAR-EVENT-REJECT')
+        .flatMap(x => {
+            const requestBody = { ...x.calendarEvent };
+
+            requestBody.status = CalendarEventStatus.Rejected;
```

Comment:
the code is the same, only status differs. Maybe have it as one method? Or, check if ofType can take two actions. If not, merge them.